aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -109,38 +125,108 @@ module aptos_framework::transaction_validation {
+
+        // Question[Orderless]: Should we move this check before checking for replay protection?
+        // Otherwise, an attacker could create a lot of transactions that fail with SEQUENCE_NUMBER_TOO_NEW.
+        // All these transactions will pass mempool validation and will be included in consensus blocks
+        // but will be discarded when the prologue is executed. The attacker can waste the space in
+        // consensus blocks without paying any cost.
+        if (!features::transaction_simulation_enhancement_enabled() || !skip_gas_payment(
+            is_simulation,
+            gas_payer_address
+        )) {
+            assert!(
+                permissioned_signer::check_permission_capacity_above(
+                    gas_payer,
+                    (max_transaction_fee as u256),
+                    GasPermission {}
+                ),
+                error::permission_denied(PROLOGUE_PERMISSIONED_GAS_LIMIT_INSUFFICIENT)
+            );
+            if (features::operations_default_to_fa_apt_store_enabled()) {
+                assert!(
+                    aptos_account::is_fungible_balance_at_least(gas_payer_address, max_transaction_fee),
+                    error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
+                );
+            } else {
+                assert!(
+                    coin::is_balance_at_least<AptosCoin>(gas_payer_address, max_transaction_fee),
+                    error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
+                );
+            }
+        }
+    }
+
+    fun check_for_replay_protection_regular_txn(
+        sender: &signer,
+        gas_payer: &signer,
+        txn_sequence_number: u64,
+        txn_authentication_key: Option<vector<u8>>,
+        is_simulation: bool,
+    ) {
         let transaction_sender = signer::address_of(sender);
         let gas_payer_address = signer::address_of(gas_payer);
-
+        // Question[Orderless]: Is this the right place to create an account if it doesn't exist?
+        if (txn_sequence_number == 0 && features::orderless_transactions_enabled()) {
+            account::create_account_if_does_not_exist(transaction_sender);
+        };
         if (
             transaction_sender == gas_payer_address
                 || account::exists_at(transaction_sender)
                 || !features::sponsored_automatic_account_creation_enabled()
                 || txn_sequence_number > 0
         ) {
-            assert!(account::exists_at(transaction_sender), error::invalid_argument(PROLOGUE_EACCOUNT_DOES_NOT_EXIST));
+            if (!features::orderless_transactions_enabled()) {
+                // Question[Orderless]: After orderless transactions is enabled, this check is not performed
+                // to accommodate stateless accounts. Is it okay?
+                assert!(account::exists_at(transaction_sender), error::invalid_argument(PROLOGUE_EACCOUNT_DOES_NOT_EXIST));
+            };
```

Comment:
we should keep this line until stateless feature is on.

Reply:
Done.